superset/assets/javascripts/dashboard/components/DashboardViewContainer.jsx

```diff
@@ -0,0 +1,335 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import urlLib from 'url';
+
+import AlertsWrapper from '../../components/AlertsWrapper';
+import GridLayout from './GridLayout';
+import Header from './Header';
+import DashboardAlert from './DashboardAlert';
+import * as Actions from '../actions';
+import { getParam } from '../../modules/utils';
+import { getExploreUrl } from '../../explore/exploreUtils';
+import { areObjectsEqual } from '../../reduxUtils';
+import { applyDefaultFormData } from '../../explore/stores/store';
+import { t } from '../../locales';
+
+import '../../../stylesheets/dashboard.css';
+
+const propTypes = {
+  actions: PropTypes.object.isRequired,
+  initMessages: PropTypes.array,
+  dashboard: PropTypes.object.isRequired,
+  datasources: PropTypes.object,
+  filters: PropTypes.object,
+  refresh: PropTypes.bool,
+  timeout: PropTypes.number,
+  user_id: PropTypes.string,
+  isStarred: PropTypes.bool,
+};
+const defaultProps = {
+  actions: {},
+  initMessages: [],
+  dashboard: {},
+  datasources: {},
+  filters: {},
+  refresh: false,
+  timeout: 60,
+  user_id: '',
+  isStarred: false,
+};
+
+class DashboardViewContainer extends React.PureComponent {
```

Comment:
I think here too is would be good to de-couple the `DashboardContainer` (manages redux, and passes props along to `Dashboard`).

Reply:
fixed. now DashboardContainer.jsx is redux only component.